File: employee_ui/src/services/employeeService.js
import { useApi } from "./useApi";

export const useEmployeeService = () => {
  const api = useApi("/employee");

  const getAllEmployees = async () => {
    const response = await api.get("/getAllEmployees");
    return response.data;
  };

  const updateEmployeeStatus = async (empCode, newStatus, updatedBy) => {
    const response = await api.put(
      `/updateEmpStatus/${empCode}?newStatus=${newStatus}&updatedBy=${updatedBy}`
    );
    return response.data;
  };

  const searchEmployees = async (payload,token) => {
    const response = await api.post("/search", payload);
    console.log(response);
    return response;
  };

  const updateEmployee = async (employeeData) => {
    const response = await api.put("/updateEmployee", employeeData);
    return response.data;
  };

  return { getAllEmployees, updateEmployeeStatus, searchEmployees, updateEmployee };
};

File: employee/src/main/java/com/example/employee/service/EmployeeService.java
package com.example.employee.service;

import java.util.List;

import com.example.employee.customException.EmployeeException;
import com.example.employee.enums.Status;
import com.example.employee.model.Employee;
import com.example.employee.model.SearchCriteria;

public interface EmployeeService {
	boolean addEmployee(Employee employee) throws EmployeeException;
	
	List<Employee> getAllEmployees() throws EmployeeException;

	boolean updateEmployee(Employee employee) throws EmployeeException;

	boolean updateEmployeeStatus(String empCode, Status newStatus, String updatedBy) throws EmployeeException;

	List<Employee> searchEmployees(SearchCriteria criteria) throws EmployeeException;
	
	Employee getEmployeeById(String emp_code) throws EmployeeException;

}
